src/js/modules/swipe.js

```diff
@@ -0,0 +1,113 @@
+
+let touchStart = {
+  x: undefined,
+  y: undefined,
+}
+let touchEnd = {
+  x: undefined,
+  y: undefined
+}
+
+/** @type {(dist: {x: number, y: number}) => boolean} */
+let touchValidator = null;
+
+/** @type {(dist: {x: number, y: number}) => void} */
+let onLeftSwipe = null
+
+/** @type {(dist: {x: number, y: number}) => void} */
+let onRightSwipe = null
+
+
+const setTouchEvents = () => {
+  window.addEventListener('touchstart', eArgs => {
+    const [touch] = eArgs.changedTouches;
+    touchStart.x = touch.screenX;
+    touchStart.y = touch.screenY
+  });
+
+  window.addEventListener('touchend', eArgs => {
+    const [touch] = eArgs.changedTouches;
+    touchEnd.x = touch.screenX;
+    touchEnd.y = touch.screenY;
+
+    const distance = {
+      x: Math.abs(touchEnd.x - touchStart.x),
+      y: Math.abs(touchEnd.y - touchStart.y),
+    }
+
+    if (touchValidator ? touchValidator.call(this, distance) : true) {
+      touchEnd.x < touchStart.x 
+        ? onLeftSwipe?.call(this, distance) 
+        : onRightSwipe?.call(this, distance)
+    }
+  });
+}
+
+/**
+ * 
+ * @param {{
+ * left: (dist: {x: number, y: number}) => void,
+ * right: (dist: {x: number, y: number}) => void,
+ * validator: (dist: {x: number, y: number}) => boolean
+ * }} configs 
+ */
+const configure = ({left, right, validator}) => {
+  if (!left || !right) throw Error('right and left swipe listeners must be defined');
+  onLeftSwipe = left;
+  onRightSwipe = right;
+  touchValidator = validator;
+  setTouchEvents();
+}
+
+export default {configure}
+
+
+
+
+
+
+// export default class SwipeHandler {
```

Comment:
Dead code

Reply:
thx, fixed in efdb6b4564d2f976ae7b4e2df0c68a8b3f2a8507